### PR TITLE
feat: make BQ client Autocloseable

### PR DESCRIPTION
**Pull request by stephaniewang526**
This will allow us to use try-with-resources in our samples.

Fixes b/145229155

### CODE REVIEW

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java [hunk 1]
```diff
@@ -37,6 +38,11 @@
   // set the option ThrowNotFound when you want to throw the exception when the value not found
   private boolean setThrowNotFound;
 
+  @Override
+  public void close() throws Exception {
+    System.out.println("Closed BigQuery Resource");
```
Comment (tswast) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java]: I imagine there are some resources to actually clean up. For example, in Python there are two long-running sockets (one for handling auth token updates, one for the actual API requests) that need to be explicitly closed.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java [hunk 2]
```diff
@@ -27,7 +27,8 @@
 import com.google.common.collect.ImmutableSet;
 import java.util.Set;
 
-public class BigQueryOptions extends ServiceOptions<BigQuery, BigQueryOptions> {
+public class BigQueryOptions extends ServiceOptions<BigQuery, BigQueryOptions>
```
Comment (tswast) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java]: Isn't it the Client class that needs to be autocloseable?

Comment (kurtisvg) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java]: +1 - this doesn't help users much.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java (hunk 1 above)
Comment (kurtisvg) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java]: +1 - Also, we probably don't want to print to `stdout`. If we do want to print this anywhere, we want to use a debug logger.

**Pull request closed without merging**